versions/modern/src/main/java/net/megavex/scoreboardlibrary/implementation/packetAdapter/modern/PacketAdapterImpl.java

```diff
@@ -87,6 +91,11 @@ public void removeSidebar(@NotNull Iterable<Player> players) {
     sendPacket(players, removePacket);
   }
 
+  @Override
+  public @NotNull ObjectivesPacketAdapter<?, ?> createObjectiveAdapter(@NotNull String objectiveName) {
+    return isNativeAdventure ? new PaperObjectivePacketAdapter(this, objectiveName) : new SpigotObjectivePacketAdapter(this, objectiveName);
```

Comment:
Using isNativeAdventure to determine if there may be some errors, as not only paperspot has Adventure libraries

Reply:
It's for making some optimisations on paper based servers

Reply:
But this cannot guarantee the existence of the class: `io.papermc.paper.adventure.AdventureComponent`, which may result in a ClassNotFoundException

Reply:
isNativeAdventure specifies whether the server is paper based, so if it is true then its safe to assume AdventureComponent exists

Reply:
I used a `custom Spigot`, which is not a branch of paper, but it also integrates the `Adventure` library and does not have the `AdventureComponent` class

Reply:
isNativeAdventure specifically checks for paper related classes so not an issue

Reply:
Oh, I'm very sorry, I didn't see the judgment on the PaperAdventure class above. I'm very sorry, I'm really sorry, it was caused by my carelessness

Reply:
its fine lol